interoperator/pkg/controller/multiclusterdeploy/sfservicebindingreplicator/sfservicebindingreplicator_controller.go

```diff
@@ -0,0 +1,360 @@
+					} else {
+						log.Error(err, "Failed to delete secret in master cluster", "binding", bindingID,
+							"clusterID ", clusterID, "state ", state)
+						// Error deleting the object - requeue the request.
+						return reconcile.Result{}, err
+					}
+				}
+			} else {
+				replicaSecret := &corev1.Secret{}
+				secretName := replica.Status.Response.SecretRef
+				if secretName == "" {
+					secretName = "sf-" + binding.GetName()
+				}
+				err = targetClient.Get(context.TODO(), types.NamespacedName{
+					Name:      secretName,
+					Namespace: replica.GetNamespace(),
+				}, replicaSecret)
+				if err != nil {
+					log.Error(err, "Failed to get secret from sister cluster", "binding", bindingID,
+						"clusterID ", clusterID, "state ", state)
+					return reconcile.Result{}, err
+				}
+				bindingSecret := &corev1.Secret{}
+				bindingSecret.Data = make(map[string][]byte)
+				bindingSecret.SetName(replicaSecret.GetName())
+				bindingSecret.SetNamespace(replicaSecret.GetNamespace())
+				for k, v := range replicaSecret.Data {
+					bindingSecret.Data[k] = v
+				}
+				if err = controllerutil.SetControllerReference(binding, bindingSecret, r.scheme); err != nil {
+					log.Error(err, "failed to set owner reference for secret", "binding", bindingID)
+					return reconcile.Result{}, err
+				}
+				log.Info("bind on sister cluster completed, replicating secret to master cluster..",
+					"clusterID", clusterID, "bindingID", bindingID, "state", state)
+				err = r.Update(context.TODO(), bindingSecret)
+				if err != nil {
+					if apiErrors.IsNotFound(err) {
+						err = r.Create(context.TODO(), bindingSecret)
+						if err != nil {
+							log.Error(err, "Error occurred while replicating secret to master cluster ",
+								"bindingID ", bindingID, "state", state)
+							return reconcile.Result{}, err
+						}
+					} else {
+						log.Error(err, "Error occurred while replicating secret to master cluster ",
+							"bindingID ", bindingID, "state", state)
+						return reconcile.Result{}, err
+					}
+				}
+			}
+		}
+		err = r.Update(context.TODO(), binding)
+		if err != nil {
+			log.Error(err, "Failed to update SFServiceBinding in master cluster", "binding", bindingID,
+				"clusterID ", clusterID, "state ", state)
+			// Error updating the object - requeue the request.
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileSFServiceBindingReplicator) setInProgress(binding *osbv1alpha1.SFServiceBinding, retryCount int) error {
+	bindingID := binding.GetName()
+	state := binding.GetState()
+	binding.SetState("in progress")
+	labels := binding.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels[constants.LastOperationKey] = state
+	binding.SetLabels(labels)
+	log.Info("Trying to update binding ", bindingID, " to state ", binding.GetState())
+	err := r.Update(context.TODO(), binding)
+	if err != nil {
+		if retryCount < constants.ErrorThreshold {
```

Comment:
[`client-go/util/retry`](https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L61) package might be worth a look for standard ways to retry.

Reply:
changed.